lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -59,6 +85,51 @@ static bool isOutputOperand(linalg::LinalgOp linalgOp, OpOperand &operand) {
   return !isInputOperand(linalgOp, operand);
 }
 
+static LogicalResult checkVNNIAccessPatterns(linalg::LinalgOp linalgOp) {
+  SmallVector<AffineMap> maps;
+  for (OpOperand &operand : linalgOp->getOpOperands()) {
+    AffineMap map = linalgOp.getMatchingIndexingMap(&operand);
+    if (isInputOperand(linalgOp, operand)) {
+      if (map.getNumResults() < 3)
+        return failure();
+      if (operand.getOperandNumber() == 1) {
+        if (map.getNumResults() != 5)
+          return failure();
+        maps.push_back(map.getMinorSubMap(4));
+      } else {
+        maps.push_back(map.getMinorSubMap(3));
+      }
+    }
+  }
+  for (OpOperand &operand : linalgOp->getOpOperands()) {
```

Comment:
You can do the same here using `linalgOp.getDpsInitOperands`. This basically avoid you checking if an operand is the input or the output.